### PR TITLE
BUG: Fix isna cannot handle ambiguous typed list

### LINKED ISSUE #20675: pandas.isnull() has poor behavior for lists
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
pd.isnull([np.NaN, 'world'])

# returns:
# array([False, False], dtype=bool)
```
#### Problem description

The output of `pd.isnull`/`pd.isna` on lists depends on the inferred dtype of the numpy conversion.
In cases where the array is inferred to be of a string type, numpy converts `np.NaN` to the string `"nan"`, which `pd.isnull()` no longer recognizes as a null value. The following solve the underlying problem, which is numpy auto-inferring a string dtype for mixed lists containing strings and `float('nan')` float values:

* explicitly convert to object arrays instead of string arrays, as is done in `pd.Series` construction
* convert to a `pd.Series` object instead of the numpy object (leverages the above)
* applying `pd.isna()` in a list comprehension for list objects, e.g. 
```python
def isna(a): #a is a list
    np.array([pd.isna(el) for el in a])
```

#### Expected Output

 ```python
array([True, False], dtype=bool)
```
or 
```python
TypeError
```
if it is undesirable to support lists with mixed float/string types with `pd.isnull()`

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.13.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-1048-aws
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: None.None

pandas: 0.22.0
pytest: 3.2.2
pip: 9.0.1
setuptools: 27.2.0
Cython: None
numpy: 1.13.3
scipy: 0.19.1
pyarrow: None
xarray: None
IPython: 5.3.0
sphinx: 1.5.4
patsy: None
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.6.0
html5lib: 0.9999999
sqlalchemy: 1.1.9
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by Licht-T**
- [x] closes #20675
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/dtypes/missing.py [hunk 1]
```diff
@@ -120,7 +120,9 @@ def _isna_new(obj):
         return _isna_ndarraylike(obj)
     elif isinstance(obj, ABCGeneric):
         return obj._constructor(obj._data.isna(func=isna))
-    elif isinstance(obj, list) or hasattr(obj, '__array__'):
+    elif isinstance(obj, list):
```
Comment (jreback) [inline pandas/core/dtypes/missing.py]: can you make the same change in _isna_old

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a whatsnew note (bug fix / reshaping)

**Comment by Licht-T** [comment]
Thanks @jreback, fixed!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @Licht-T

**Pull request merged**